corva/middleware/unpack_context.py

```diff
@@ -0,0 +1,26 @@
+from typing import Callable
+
+from corva.models.base import BaseContext
+
+
+def unpack_context_factory(include_state=False, include_context=False):
```

Comment:
I still don't get the purpose of this function. Better to use different functions to process different tasks types

Reply:
reworked the logic